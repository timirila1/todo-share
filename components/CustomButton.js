import React from 'react';
import { StyleSheet, TouchableOpacity, Text } from 'react-native';
import PropTypes from 'prop-types';

import { colors } from '../services/consts';

export const CustomButton = (props) => {
    const {
        title,
        onPress,
        mod,
        disabled,
    } = props;

    return (
        <TouchableOpacity
            style={[
                styles.button,
                mod === 'light' && styles.lightButton,
                mod === 'trans' && styles.transButton,

            ]}
            disabled={disabled}
            onPress={onPress}
        >
            <Text
                style={[
                    styles.title,
                    mod === 'light' && styles.lightModText,
                    mod === 'trans' && styles.transModText,
                ]}>
                {title}
            </Text>
        </TouchableOpacity>
    );
};

const styles = StyleSheet.create({
    button: {
        backgroundColor: colors.additional,
        padding: 10,
        borderRadius: 10,
    },
    disabledButton: {
        opacity: 0.8,
    },
    lightButton: {
        backgroundColor: 'white',
    },
    transButton: {
        backgroundColor: 'rgba(255, 255, 255, 0)',
    },
    title: {
        textAlign: 'center',
        color: 'white',
        fontSize: 24,
        fontWeight: '300',
        lineHeight: 36,
    },
    lightModText: {
        color: colors.additional,
    },
    transModText: {
        color: 'white',
    },
});

CustomButton.propTypes = {
    title: PropTypes.string,
    onPress: PropTypes.func,
    mod: PropTypes.string,
    disabled: PropTypes.bool,
};
