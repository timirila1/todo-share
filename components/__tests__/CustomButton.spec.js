import 'react-native';
import React from 'react';
import { CustomButton } from '../CustomButton';
import renderer from 'react-test-renderer';

it('Render BoardsFooter', async () => {
    const tree = renderer.create(
        <CustomButton title="test" />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
