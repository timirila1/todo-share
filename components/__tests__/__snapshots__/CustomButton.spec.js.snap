// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render BoardsFooter 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#fa6c35",
      "borderRadius": 10,
      "opacity": 1,
      "padding": 10,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "white",
          "fontSize": 24,
          "fontWeight": "300",
          "lineHeight": 36,
          "textAlign": "center",
        },
        false,
        false,
      ]
    }
  >
    test
  </Text>
</View>
`;
